.github/workflows/unit_tests.yaml

```diff
@@ -1,103 +1,24 @@
 name: linux tests
 
 on:
-  push:
-    branches:
-      - develop
-      - releases/**
-  pull_request:
-    branches:
-      - develop
-      - releases/**
+  workflow_run:
+    workflows: [style and validate]
+    types:
+      - completed
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_number }}
   cancel-in-progress: true
 
 jobs:
-  # Validate that the code can be run on all the Python versions
-  # supported by Spack
-  validate:
+  style-check:
```

Comment:
I think this should be removed entirely (vs. right now it essentially appears to be a noop). In that case, there is no longer a need for a `needs:` section to many of these checks.

Reply:
From what I understood from the GHA docs, this is a required workaround to achieve the workflow we want from the proposed setup of workflow files. In order to have one workflow depend on another in GHA, the `on` trigger should be set off by the run of a workflow (the style checks). If we then want the run of the subsequent workflow (the unit tests) to be dependent not only on the completion of the initial workflow, but the success of the contents, we need a 'job' in the dependent workflow to check for success of the completed job. I think this job could be cleaned up/clarified, but we cannot discard it entirely. 

That being said I may have missed something in my pass over the docs. I'll look again and update my comment if need be.